pkg/serviceproviders/application/helper.go

```
@@ -116,13 +117,32 @@ func compileResource(obj runtime.Object, params ResourceParams) (runtime.Object,
 		New("document").
 		Funcs(template.FuncMap{
 			"json": func(v interface{}) interface{} {
-				val, _ := json.Marshal(v)
```

The template functions can return an error as a second argument, so we shouldn't swallow them.

---

done, see 2beef6c